test/suites/integration/bsp/submit-proofs.test.ts

```diff
@@ -242,7 +242,49 @@ describeBspNet(
       // TODO: Check that BSP-Three no longer has a challenge deadline.
     });
 
-    it("BSP submits proof, transaction gets dropped, BSP-resubmits and succeeds", async () => {
+    it(
+      "BSP submits proof, transaction gets dropped, BSP-resubmits and succeeds",
+      { skip: "Not implemented yet." },
+      async () => {
+        const lastTickResult = await userApi.call.proofsDealerApi.getLastTickProviderSubmittedProof(
+          userApi.shConsts.DUMMY_BSP_ID
+        );
+        const lastTickBspSubmittedProof = lastTickResult.asOk.toNumber();
+        const challengePeriodResult = await userApi.call.proofsDealerApi.getChallengePeriod(
+          userApi.shConsts.DUMMY_BSP_ID
+        );
+        const challengePeriod = challengePeriodResult.asOk.toNumber();
+        const nextChallengeTick = lastTickBspSubmittedProof + challengePeriod;
+        await userApi.advanceToBlock(nextChallengeTick);
+        await userApi.block.seal({ finaliseBlock: false });
+
+        await userApi.assert.extrinsicPresent({
+          module: "proofsDealer",
+          method: "submitProof"
+        });
+        await userApi.block.reOrg();
+
+        await assert.rejects(
+          async () => {
+            await userApi.assert.extrinsicPresent({
+              module: "proofsDealer",
+              method: "submitProof",
+              timeout: 1000
+            });
+          },
+          /No matching extrinsic found for proofsDealer\.submitProof/,
+          "No submit proof extrinsics after re-org"
+        );
+
+        await userApi.block.seal();
+        await userApi.assert.extrinsicPresent({
+          module: "proofsDealer",
+          method: "submitProof"
+        });
+      }
+    );
+
+    it("Proofs not submitted in unfinalised block", { skip: "Not implemented yet." }, async () => {
```

Comment:
I think there's been a confusion, and I'm likely to blame. BSPs _should_ submit proofs even if blocks are not finalised. Otherwise, if finality is lagging and they don't submit proofs, once finality catches up, they might already be late to submit it and be slashed.